drivers/spi/spi_ll_stm32.h

```diff
@@ -25,57 +25,126 @@ struct spi_stm32_data {
 
 static inline u32_t spi_stm32_ll_func_IsActiveFlag_TXE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	return LL_SPI_IsActiveFlag_TXP(spi);
+#else
 	return LL_SPI_IsActiveFlag_TXE(spi);
+#endif
 }
 
 static inline u32_t spi_stm32_ll_func_IsActiveFlag_RXNE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	return LL_SPI_IsActiveFlag_RXP(spi);
+#else
 	return LL_SPI_IsActiveFlag_RXNE(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_EnableIT_TXE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_TXP(spi);
+#else
 	LL_SPI_EnableIT_TXE(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_EnableIT_RXNE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_RXP(spi);
+#else
 	LL_SPI_EnableIT_RXNE(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_EnableIT_ERR(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_EnableIT_UDR(spi);
+	LL_SPI_EnableIT_OVR(spi);
+	LL_SPI_EnableIT_CRCERR(spi);
+	LL_SPI_EnableIT_FRE(spi);
+	LL_SPI_EnableIT_MODF(spi);
+#else
 	LL_SPI_EnableIT_ERR(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_DisableIT_TXE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_TXP(spi);
+#else
 	LL_SPI_DisableIT_TXE(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_DisableIT_RXNE(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_RXP(spi);
+#else
 	LL_SPI_DisableIT_RXNE(spi);
+#endif
 }
 
 static inline void spi_stm32_ll_func_DisableIT_ERR(SPI_TypeDef *spi)
 {
+#ifdef CONFIG_SOC_SERIES_STM32MP1X
+	LL_SPI_DisableIT_UDR(spi);
+	LL_SPI_DisableIT_OVR(spi);
+	LL_SPI_DisableIT_CRCERR(spi);
+	LL_SPI_DisableIT_FRE(spi);
+	LL_SPI_DisableIT_MODF(spi);
```

Comment:
I see all this changes for MP1. But is it really necessary to do it at zephyr level?
Shouldn't be a better idea to incorporate the SoC difference in LL driver?
In this case we will always call LL_SPI_DisableIT_ERR() which will do the proper thing for MP1.

Reply:
@avisconti, this is a separate topic that should be discussed internally.